adaptdl_cli/bin/adaptdl

```diff
@@ -17,86 +17,115 @@
-from adaptdl_cli.registry import fix_local_docker, fix_etc_hosts, \
-    ADAPTDL_REGISTRY_URL
 
 
 def _get_registry_info():
-    if 'ADAPTDL_SUBMIT_REPO' and 'ADAPTDL_SUBMIT_REPO_CREDS' in os.environ:
+    if 'ADAPTDL_SUBMIT_REPO' in os.environ:
         print(f"Using external repo at "
               f"{os.environ['ADAPTDL_SUBMIT_REPO']}")
-        return os.environ['ADAPTDL_SUBMIT_REPO'], \
-            os.environ['ADAPTDL_SUBMIT_REPO_CREDS'], False
+        return (os.environ['ADAPTDL_SUBMIT_REPO'],
+                os.getenv('ADAPTDL_SUBMIT_REPO_CREDS'))
     else:
-        print(f"Using AdaptDL internal registry at {ADAPTDL_REGISTRY_URL}")
-        return ADAPTDL_REGISTRY_URL, None, True
-
-
-def _prefix(image):
-    # RepoDigests have sha256 suffixes and not user tags, so we remove it here
-    parts = image.split('/')
-    return '/'.join(parts[:-1] + [parts[-1].split(':')[0]])
-
-
-def _replace_localhost(repodigest):
-    return ":".join(["localhost"] + repodigest.split(':')[1:])
-
-
-def submit(args, remaining):
-    repo, secret, internal = _get_registry_info()
-    if internal:
-        # We choose the full reponame
-        image = f"{repo}/dev/adaptdl-submit:latest"
+        print("Using AdaptDL internal registry")
+        return None, None
+
+
+def _find_insecure_registry():
+    services = kubernetes.client.CoreV1Api().list_service_for_all_namespaces(
+        label_selector="app=docker-registry",
+        field_selector="metadata.name=adaptdl-registry")
+    for service in services.items:
+        for port in service.spec.ports:
+            if port.node_port and port.name == "registry":
+                return (service.metadata.namespace,
+                        service.metadata.name + ":" + port.name,
+                        port.node_port)
+    return None, None, None
+
+
+def _get_insecure_registry_proxy(proxy_port):
+    if platform.system() == "Darwin":
+        # Docker for Mac runs in a VM and can't access the proxy which is bound
+        # to a port on the host. Use a special DNS name instead.
+        # https://github.com/docker/for-mac/issues/3611
+        # https://docs.docker.com/docker-for-mac/networking/#use-cases-and-workarounds
+        registry = f"host.docker.internal:{proxy_port}"
+        # Docker doesn't automatically trust host.docker.internal as an
+        # insecure registry. Need to add it manually.
+        docker_dir = Path.home() / ".docker"
+        daemon_path = docker_dir / "daemon.json"
+        try:
+            with open(daemon_path, "r") as f:
+                daemon = json.load(f)
+        except FileNotFoundError:
+            daemon = {}
+        if registry not in daemon.setdefault("insecure-registries", []):
+            daemon["insecure-registries"].append(registry)
+            docker_dir.mkdir(exist_ok=True)
+            with open(daemon_path, "w") as f:
+                json.dump(daemon, f, indent=2)
+            raise SystemExit(f"Insecure registry {registry} written to"
```

Comment:
If the user forgets to restart docker or doesn't do it. Would this run silently on the his next try?

Reply:
It will fail at the docker push step

Reply:
yes but from the next try onwards users won't see this message and would never know what they missed (?)

Reply:
The next time, the following error is printed:
```
The push refers to repository [host.docker.internal:55555/adaptdl-submit]
Get https://host.docker.internal:55555/v2/: x509: certificate signed by unknown authority
Error: could not push image host.docker.internal:55555/adaptdl-submit
```
Googling for "docker x509: certificate signed by unknown authority" returns as the first result a stackoverflow answer suggesting to restart docker. I think that's fine unless there's a better way for us to detect that a Docker restart is needed.

Reply:
I think we should document this as part of some FAQs

Reply:
Added